scripts/normalize-by-median.py

```diff
@@ -373,28 +293,48 @@ def main():  # pylint: disable=too-many-branches,too-many-statements
 
     input_filename = None
 
+    # diginorm algorithm lives in Normalizer, go get it
     norm = Normalizer(args.cutoff, htable, report_fp)
 
-    for f in CheckpointCountingTable(args.input_filenames, args.dump_frequency,
-                                     htable, args.savetable):
-        total_acc, discarded_acc, corrupt = \
-            normalize_by_median_and_check(
-                f, htable, args.single_output_file,
-                args.fail_save, args.paired, args.force, norm, report_fp)
-        corrupt_files += corrupt
-
-    # Stuff to handle paired and unpaired data
-    if args.paired and args.unpaired_reads:
-        args.paired = False
-        output_name = args.unpaired_reads
+    # make a list of all filenames and if they're paired or not
+    # if we don't know if they're paired, default to not forcing paired
+    files = []
+    for e in args.input_filenames:
+        files.append([e, args.paired])
+    if args.unpaired_reads:
+        files.append([args.unpaired_reads, False])
+
+    corrupt_files = []
+
+    outfp = None
+
+    if args.single_output_file:
+        if args.single_output_file is sys.stdout:
+            output_name = '/dev/stdout'
+        else:
+            output_name = args.single_output_file.name
+        outfp = args.single_output_file
+
+    for f, p in files:
         if not args.single_output_file:
-            output_name = os.path.basename(args.unpaired_reads) + '.keep'
-        outfp = open(output_name, 'w')
-        total_acc, discarded_acc, corrupt = \
-            normalize_by_median_and_check(
-                args.unpaired_reads, htable, args.single_output_file,
-                args.fail_save, args.paired, args.force, norm, report_fp)
-        corrupt_files += corrupt
+            output_name = os.path.basename(f) + '.keep'
+            outfp = open(output_name, 'w')
+
+        ksize = htable.ksize()
```

Comment:
This can be moved up to where we create/load htable; not needed inside the loop.